variables.tf

```diff
@@ -388,6 +388,14 @@ variable "cluster_name" {
   description = "(Optional) The name for the AKS resources created in the specified Azure Resource Group. This variable overwrites the 'prefix' var (The 'prefix' var will still be applied to the dns_prefix if it is set)"
 }
 
+variable "confidential_computing" {
+  type = list(object({
```

Comment:
According to the [schema](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/kubernetes_addons.go#L89-L101), could we set type to `boolean` for this variable?

Reply:
Yes, we can simplify this type to a `boolean`, I was wondering there might be a possibility that more attributes will be added to this `confidential_computing` in the future, use a `list` would be more scalable?

Reply:
Since the `MaxItems` of this list is `1` I think in that case, an `object` would be enough for us. WDYT?

Reply:
Make sense. We can just use `object` instead of `list`.